src/app/core/component/graph-display/graph-display.component.ts

```diff
@@ -162,19 +186,33 @@ export class GraphDisplayComponent implements AfterContentInit, OnChanges {
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.data != null) {
       this.isLoading = true;
+      this.initData();
       this.getGraph();
     }
+    if (changes.showConcreteSolutions != null) {
+      setTimeout(() => { // Give time for property updates
+        this.triggerRerendering(true);
+      });
+    }
   }
 
   edgeAdded(event) {
```

Comment:
A function does something, i.e., a name should indicate this. Although this is an event listener it is much more readable of the name indicates what's going on: e.g., handleEdgeAddedEvent might be better.

Reply:
Done